### PR TITLE
Update Tagged SVG Output to use ModelTagGroups

### PR DESCRIPTION
**Pull request by eslickj**
This mainly updates updates the svg_tag function to use the new model tag classes.  It also move some feature (conditional formatting, division by zero trapping, and no value trapping) to model tags, as it was previously part of the svg output function.

I moved the svg_tag function and updated the API.  The new function support the old API and old import location.  At some future point we can add deprecation warnings.  I don't want to bother anyone with it yet though. 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/core/util/misc.py [hunk 1]
```diff
@@ -117,114 +117,6 @@ def TagReference(s, description=""):
     return r
 
 
-# Author John Eslick
```
Comment (andrewlee94) [inline idaes/core/util/misc.py:120]: I think we should leave this in for now with a deprecation warning. You never know who might be using this.

Edit: It looks like you moved this to the `tags` module - you should add a deprecation warning with a redirect here then.

File: idaes/core/util/tags.py [hunk 2]
```diff
@@ -223,12 +242,6 @@ def get_unit_str(self, index=None):
     def is_var(self):
         """Whether the tagged expression is a Pyomo Var. Tagged variables
         can be fixed or set, while expressions cannot.
-
```
Comment (andrewlee94) [inline idaes/core/util/tags.py:226]: Why did you delete the documentation here? It is rare for more documentation to be bad thing.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
Only a couple of minor comments, the main one being to add a deprecation warning and redirect for the old svg tag method.

File: idaes/core/util/tags.py (hunk 2 above)
Comment (eslickj) [reply idaes/core/util/tags.py:226]: It's a property.  It used to be a function, but I forgot to fix the docstring.

Comment (andrewlee94) [reply idaes/core/util/tags.py:226]: Ah - Github conveniently cut off the display right at the `def` line so I didn't see that.

File: idaes/core/util/misc.py (hunk 1 above)
Comment (eslickj) [reply idaes/core/util/misc.py:120]: Added deprecation warnings for both the old location and the old API.

**Comment by eslickj** [comment]
Ok, should have comments fixed and I fixed the doc example problem that was breaking tests.

File: idaes/core/util/misc.py [hunk 3]
```diff
@@ -132,6 +131,16 @@ def copy_port_values(destination=None, source=None, arc=None,
             direction=direction)
 
 
+def svg_tag(*args, **kwargs):
+    """
+    Moved to idaes.core.util.tags.svg_tag
+    Leaving redirection function here for deprecation warning.
+    """
+    _log.warning("DEPRECATED: idaes.core.util.misc.svg_tag has been deprecated. "
```
Comment (andrewlee94) [inline idaes/core/util/misc.py:139]: I would suggest using the Pyomo deprecation warning here: from pyomo.common.deprecation import deprecation_warning

This lets you tag the version that is was deprecated in and the version it will be removed in.

Otherwise, I think you should make a note of when this deprecation was added so we can judge when to finally remove it.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
A couple of minor typos (mostly so minor I almost ignored them), but I think we need to be more rigorous with deprecations and record when they were added (and if possible, when they will be removed).

**Comment by eslickj** [comment]
Fixed typos and now use Pyomo deprecation stuff.

**Review by andrewlee94 — APPROVED** [review]

**Review by dallan-keylogic — APPROVED** [review]
